lib/sisimai/bite/email/x1.rb

```diff
@@ -0,0 +1,124 @@
+  module X1
+    # Sisimai::Bite::Email::X1 parses a bounce email which created by Unknown
+    # MTA #1. Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/X1.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :from    => %r/["]Mail Deliver System["] /,
+        :subject => %r/\AReturned Mail: /,
+      }
+      Re1 = {
+        :begin   => %r/\AThe original message was received at (.+)\z/,
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+        :rfc822  => %r/\AReceived: from \d+[.]\d+[.]\d+[.]\d/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Unknown MTA #1'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return []; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Unknown MTA #1
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['subject'] =~ Re0[:subject]
+        return nil unless mhead['from']    =~ Re0[:from]
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        datestring = ''     # (String) Date string
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:'message-rfc822']).zero?` instead of `readcursor & Indicators[:'message-rfc822'] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->